integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/v2/ContentDisposition.java

```diff
@@ -0,0 +1,261 @@
+            String part = parts.get(i);
+            int eqIndex = part.indexOf('=');
+            if (eqIndex != -1) {
+                String attribute = part.substring(0, eqIndex);
+                String value = (part.startsWith("\"", eqIndex + 1) && part.endsWith("\"") ?
+                        part.substring(eqIndex + 2, part.length() - 1) :
+                        part.substring(eqIndex + 1, part.length()));
+                if (attribute.equals("name")) {
+                    name = value;
+                } else if (attribute.equals("filename*")) {
+                    filename = decodeHeaderFieldParam(value);
+                    charset = Charset.forName(value.substring(0, value.indexOf('\'')));
+                    Assert.isTrue(UTF_8.equals(charset) || ISO_8859_1.equals(charset),
+                            "Charset should be UTF-8 or ISO-8859-1");
+                } else if (attribute.equals("filename") && (filename == null)) {
+                    filename = value;
+                } else if (attribute.equals("size")) {
+                    size = Long.parseLong(value);
+                } else if (attribute.equals("creation-date")) {
+                    try {
+                        creationDate = OffsetDateTime.parse(value, RFC_1123_DATE_TIME);
+                    } catch (DateTimeParseException ex) {
+                        // ignore
+                    }
+                } else if (attribute.equals("modification-date")) {
+                    try {
+                        modificationDate = OffsetDateTime.parse(value, RFC_1123_DATE_TIME);
+                    } catch (DateTimeParseException ex) {
+                        // ignore
+                    }
+                } else if (attribute.equals("read-date")) {
+                    try {
+                        readDate = OffsetDateTime.parse(value, RFC_1123_DATE_TIME);
+                    } catch (DateTimeParseException ex) {
+                        // ignore
+                    }
+                }
+            } else {
+                throw new IllegalArgumentException("Invalid content disposition format");
+            }
+        }
+        return new ContentDisposition(type, name, filename, charset, size, creationDate, modificationDate, readDate);
+    }
+
+    private static List<String> tokenize(String headerValue) {
+        int index = headerValue.indexOf(';');
+        String type = (index >= 0 ? headerValue.substring(0, index) : headerValue).trim();
+        if (type.isEmpty()) {
+            throw new IllegalArgumentException("Content-Disposition header must not be empty");
+        }
+        List<String> parts = new ArrayList<>();
+        parts.add(type);
+        if (index >= 0) {
+            do {
+                int nextIndex = index + 1;
+                boolean quoted = false;
+                while (nextIndex < headerValue.length()) {
+                    char ch = headerValue.charAt(nextIndex);
+                    if (ch == ';') {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS134)